Tests/SwiftParserTest/DeclarationTests.swift

```diff
@@ -1427,6 +1427,208 @@ final class DeclarationTests: XCTestCase {
     )
   }
 
+  func testAttributedMacroExpansionDeclaration() {
+    assertParse(
```

Comment:
Could you add separate `assertParse` calls for these three cases? It makes it a lot easier to understand and debug the test cases IMO.

Same for all the other test cases, could you only have a single macro expansion in each `assertParse` function? That way the `substructure` can also be reduced to the `MacroExpansionDeclSyntax` node.